integration-tests/src/test/java/oracle/kubernetes/operator/ItSitConfigDomainInImage.java

```diff
@@ -0,0 +1,277 @@
+
+  private static String testClassName;
+  private static int testNumber;
+  private static Operator operator1;
+  private static Domain domain;
+  private static String sitconfigTmpDir = "";
+  private static String mysqltmpDir = "";
+  private static String configOverrideDir = "";
+  private static String mysqlYamlFile = "";
+  private static String domainNS;
+  private static String testprefix = "sitconfigdomaininimage";
+  private static String mysqldbport;
+  private static String adminpodname;
+  private static String JDBC_URL;
+  private static StringBuffer namespaceList;
+  /**
+   * This method gets called only once before any of the test methods are executed. It does the
+   * initialization of the integration test properties defined in OperatorIT.properties and setting
+   * the resultRoot, pvRoot and projectRoot attributes.
+   *
+   * @throws Exception when the initialization, creating directories , copying files and domain
+   *                   creation fails.
+   */
+
+  @BeforeClass
+  public static void staticPrepare() throws Exception {
+    if (FULLTEST) {
+      testClassName = new Object() {
+      }.getClass().getEnclosingClass().getSimpleName();
+      initialize(APP_PROPS_FILE, testClassName);
+      TEST_RES_DIR = getProjectRoot() + "/integration-tests/src/test/resources/";
+      testNumber = getNewSuffixCount();
+    }
+  }
+
+  /**
+   * This method gets called only once before any of the test methods are executed. It does the
+   * initialization of the integration test properties defined in OperatorIT.properties and setting
+   * the resultRoot, pvRoot and projectRoot attributes.
+   *
+   * @throws Exception when the initialization, creating directories , copying files and domain
+   *                   creation fails.
+   */
+  @Before
+  public void prepare() throws Exception {
+    // initialize test properties and create the directories
+    if (FULLTEST) {
+      // create operator1
+      if (operator1 == null) {
+        createResultAndPvDirs(testClassName);
+        Map<String, Object> operatorMap = createOperatorMap(testNumber, true, testprefix);
+        operator1 = TestUtils.createOperator(operatorMap, Operator.RestCertType.SELF_SIGNED);
+        Assert.assertNotNull(operator1);
+        domainNS = ((ArrayList<String>) operatorMap.get("domainNamespaces")).get(0);
+        namespaceList = new StringBuffer((String) operatorMap.get("namespace"));
+        namespaceList.append(" ").append(domainNS);
+        sitconfigTmpDir = getResultDir() + "/sitconfigtemp" + testprefix;
+        mysqltmpDir = sitconfigTmpDir + "/mysql";
+        configOverrideDir = sitconfigTmpDir + "/configoverridefiles";
+        mysqlYamlFile = mysqltmpDir + "/mysql-dbservices.yml";
+
+        Files.createDirectories(Paths.get(sitconfigTmpDir));
+        Files.createDirectories(Paths.get(configOverrideDir));
+        Files.createDirectories(Paths.get(mysqltmpDir));
+        mysqldbport = String.valueOf(31306 + testNumber);
```

Comment:
create a constant variable for mysql db base port 31306 ...best place to generate port numbers is BaseTest.createDomainMap() but they may be too many changes for you ...

Reply:
mysql is independent from domain creation, will keep separate, port number is generated as 31306 + getNewSuffixNumber()